### PR TITLE
Separating `list` from `read` function and other small tweaks.

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Rename `Faker.string_value` to `Faker.text_value`.
- Remove `Regex.pattern_string` as duplicate of `Regex.pattern`.
- Sort the Date picker.
- Rename `Data.list_directory` to `Data.list`.
- Remove support for reading a directory.

image

image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Review by marthasharkey — APPROVED** [review]

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso [hunk 1]
```diff
@@ -868,11 +868,12 @@ Date.from (that:JS_Object) =
...
-private make_day_picker cache=Nothing =
+private make_day_picker value cache =
+    _ = value
     year = cache.if_not_nothing <| cache "year"
     month = cache.if_not_nothing <| cache "month"
     days = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
     max = if month.is_nothing then 31 else
         if month != 2 || year.is_nothing then days.at (month - 1) else
-            if Date.new year month 1 . is_leap_year then 29 else 28
+            if Date.new year 1 1 . is_leap_year then 29 else 28
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:878]: btw. why this change? It seems like both old and new code were doing the same thing anyway?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_File.enso [hunk 2]
```diff
@@ -259,7 +259,7 @@ type Enso_File
                 json = Utils.http_request_as_json HTTP_Method.Get asset.internal_uri
                 datalink = Data_Link_Helpers.interpret_json_as_data_link json
                 datalink.read format on_problems
-            Enso_Asset_Type.Directory -> if format == Auto_Detect then self.list else Error.throw (Illegal_Argument.Error "Directories can only be read using the Auto_Detect format.")
+            Enso_Asset_Type.Directory -> Error.throw (Illegal_Argument.Error "Cannot `read` a directory, use `list`.")
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_File.enso:262]: This is much better

File: distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Existing_Enso_Asset.enso [hunk 3]
```diff
@@ -98,7 +98,7 @@ type Existing_Enso_Asset
     is_regular_file self = self.asset_type == Enso_Asset_Type.File
 
     ## PRIVATE
-    list_directory self =
+    list self =
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Existing_Enso_Asset.enso]: This was internal API so IMO did not have to be renamed.

But renaming seems OK too. Feels like `list` is nice for consistency, whereas `list_directory` is nice for explicit-ness in the internal API. Both have a + and - so either seems fine.

File: distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Postgres/Postgres_Dialect.enso [hunk 4]
```diff
@@ -596,18 +596,18 @@ replace args metadata =
                     _ ->
                         SQL_Builder.code "REGEXP_REPLACE(" ++ input ++ ", " ++ escaped_pattern ++ ", " ++ replacement ++ ")"
         Regex ->
-            pattern_string = SQL_Builder.interpolation raw_pattern.pattern_string
+            pattern = SQL_Builder.interpolation raw_pattern.pattern
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Postgres/Postgres_Dialect.enso]: tbh I actually preferred the old way here.
This is internals - we don't need to care about user-facing naming here.

The `pattern_string` makes it clear this is a string/text value. The `pattern` is unclear - is it a string or is it a `Regex.Pattern` instance, or something?

If you want to avoid _string_ even in internals code, I'd at least rename this variable to `pattern_text`, to make it clear that this is the text representation of the pattern.

File: test/Table_Tests/src/IO/Delimited_Read_Spec.enso [hunk 5]
```diff
@@ -95,11 +95,6 @@ add_specs suite_builder =
             r1.should_fail_with File_Error
             r1.catch.should_be_a File_Error.Not_Found
 
-            directory = enso_project.data
-            r2 = Data.read directory (..Delimited "," headers=True value_formatter=Nothing) Problem_Behavior.Report_Error
-            r2.should_fail_with File_Error
-            r2.catch.should_be_a File_Error.IO_Error
```
Comment (radeusgd) [inline test/Table_Tests/src/IO/Delimited_Read_Spec.enso]: Why is this test removed?

**Review by radeusgd — APPROVED** [review]
Looks good, just a few small questions and one remark on naming (the `pattern` case).

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso (hunk 1 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:878]: Just didn't seem necessary to feed this into the Date.

File: test/Table_Tests/src/IO/Delimited_Read_Spec.enso (hunk 5 above)
Comment (jdunkerley) [reply test/Table_Tests/src/IO/Delimited_Read_Spec.enso]: Will restore and updated version.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:878]: Fair
